### PR TITLE
Merge changes on dev

### PR DESCRIPTION
**Pull request by bl4szk4**

### CODE REVIEW

File: Ellipse.dxf
Comment (bl4szk4) [inline Ellipse.dxf:1]: This file must be deleted before merge

File: app/consts/presentation_consts.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+new_scale = range(1, 101)
+
+min_val = math.log10(0.01)
+max_val = math.log10(100)
+
+SCALE_MAPPING = {
+    i: 10 ** (max_val - (i - 1) * (max_val - min_val) / 99)
+    for i in new_scale
+}
```
Comment (bl4szk4) [inline app/consts/presentation_consts.py:26]: Maybe we should think about moving it somewhere else, as it is not good to make that operations in consts, but IDK if that makes sence.

File: app/files_processing/file_reading.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    def get_figures(self) -> List:
+        msp = self.__dxf.modelspace()
+
+        figures = [self.get_coordinates(entity) for entity in msp]
+        return figures
+
+    @staticmethod
+    def get_coordinates(entity) -> Entity:
+        # for entity in msp:
```
Comment (bl4szk4) [inline app/files_processing/file_reading.py]: Leftover

File: app/files_processing/file_reading.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        entity_type = entity.dxftype()
+
+        match entity_type:
+            case Figures.LINE.value:
+                # print([(entity.dxf.start.x, entity.dxf.start.y,), (entity.dxf.end.x, entity.dxf.end.y,)])
```
Comment (bl4szk4) [inline app/files_processing/file_reading.py]: Leftover

File: app/files_processing/file_reading.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                return Entity(coords=[(v.dxf.location.x, v.dxf.location.y) for v in entity.vertices],
+                              entity_type=Figures.POLYLINE)
+
+            case Figures.LWPOLYLINE.value:
+                point, start_width, end_width, bulge = entity
+                print(f"{point=}, {start_width=}, {end_width=}, {bulge=}")
+                # spline_curve = entity.construction_tool()
+                # points = list(spline_curve.flattening(0.01))
+                # return Entity(coords=[(p.x, p.y,) for p in points],
+                #               entity_type=Figures.LWPOLYLINE)
```
Comment (bl4szk4) [inline app/files_processing/file_reading.py]: Leftover

File: app/files_processing/file_reading.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            case Figures.SPLINE.value:
+                control_points = np.array(entity.control_points)
+                control_points = control_points[:, :2]
+
+                return Entity(coords=control_points, entity_type=Figures.SPLINE)
+            case _:
+                print(entity_type)
```
Comment (bl4szk4) [inline app/files_processing/file_reading.py]: Change to log.

File: app/presentation/components/canvas/basic_canvas.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    CANVAS_WIDTH,
+    CANVAS_HEIGHT,
+    CANVAS_PEN_COLOR,
+    GRAY_COLOR,
+    GRID_WIDTH,
+    GRID_STEP,
+)
+
+
+class BasicCanvas(QtWidgets.QWidget):
```
Comment (bl4szk4) [inline app/presentation/components/canvas/basic_canvas.py:16]: IMO as it is "abstract class" I should have used `ABC` class, but it is just programming mistake.

File: app/presentation/components/slider.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QVBoxLayout, QSlider, QLabel
+from app.consts.presentation_consts import SLIDER_MAX, SLIDER_MIN, SLIDER_TIC
+
+
+class Slider(QVBoxLayout):
+    def __init__(self):
+        super().__init__()
+        self.print_scale_slider: QSlider = QSlider(Qt.Horizontal)
+        self.print_scale_slider_label: QLabel = QLabel("Scale 1")
```
Comment (bl4szk4) [inline app/presentation/components/slider.py]: I think that we can get rid of that text field, as right now it doesn't say anything to the user

File: app/presentation/panels/main_panel.py [hunk 8]
```diff
@@ -118,27 +128,47 @@ def handle_connection_prior(self, connect: Callable[[str], ServiceError]):
...
         )
         self.port_coms_grid.button_connect_laser.clicked.connect(
             lambda: self.handle_connection_laser(laser_init)
         )
-        self.stage_info_grid.start_timer(stage_info) # TODO think later about how to run it in different thread
+
+        self.stage_management_grid.button_load_file.clicked.connect(lambda x: WindowUtils.open_file(self))
 
     @staticmethod
     def show_notification(title: str = "SUCCESS",
```
Comment (bl4szk4) [inline app/presentation/panels/main_panel.py:171]: There is a huge problem with notifications, as whem they appear they clear all canvas.

File: app/presentation/panels/processing_panel.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QWidget, QHBoxLayout, QApplication
+
+from app.presentation.components.canvas.basic_canvas import BasicCanvas
+
+
+class ProcessingPanel(QWidget):
```
Comment (bl4szk4) [inline app/presentation/panels/processing_panel.py:7]: Right now it is unused, so I don't know if it is really necessary

File: app/presentation/services/canvas_drawing.py [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+    @staticmethod
+    def draw_arc(painter: QPainter, coords: list, radius: float, params: tuple) -> None:
+
+        center = WindowUtils.convert_float_to_int_list(coords[0])
+        start_angle, end_angle = params
+        radius = int(radius)
+
+        start_angle_degrees = start_angle * 16
+        end_angle_degrees = end_angle * 16
```
Comment (bl4szk4) [inline app/presentation/services/canvas_drawing.py:85]: I guess this should be const

File: app/presentation/services/figures_transition.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
+from app.files_processing.enums import Figures
+from app.files_processing.models import Entity
+from app.consts.presentation_consts import PRESENTATION_OFFSET
+import math
+
+
+class FiguresTransitionService:
```
Comment (bl4szk4) [inline app/presentation/services/figures_transition.py:7]: this class without any comments is really hard to read and understand XD

File: app/presentation/window_utils/window_utils.py [hunk 12]
```diff
@@ -12,15 +14,70 @@ def open_file(main_window):
...
+    @staticmethod
+    def convert_list_of_list_float_to_int(float_list: list[tuple[float]]) -> list[tuple[int, int]]:
+        converted = [(int(x), int(y)) for x, y in float_list]
+        return converted
+
+    @staticmethod
+    def get_offset(entities: list[Entity]) -> tuple[int, int]:
+        min_x = entities[0].coords[0][0]
+        min_y = entities[0].coords[0][1]
+        # print(f"{min_x} --- {min_y}")
```
Comment (bl4szk4) [inline app/presentation/window_utils/window_utils.py:33]: Leftover

File: app/service/service.py [hunk 13]
```diff
@@ -163,10 +168,12 @@ def check_position(self):
...
 
     # TODO: proper laser switching and consider triggers
-    def print_lines(self, lines: List[List[Tuple[int, int]]], dxf_file_path: str, from_canvas: bool):
+    def print_lines(
+            self, lines: List[List[Tuple[int, int]]], dxf_figures: List[Entity], from_canvas: bool, scale: int
+    ):
         if from_canvas:
-            scaled_lines = [StageUtils.scale_list_points(line, self.__service_app_params.scale_x,
-                                                         self.__service_app_params.scale_y) for line in lines]
+            # FIXME: do we need this scale?
```
Comment (bl4szk4) [inline app/service/service.py:169]: Do we?

File: app/service/service.py [hunk 14]
```diff
@@ -271,3 +273,18 @@ def get_stage_info(self) -> List:
         else:
             x, y = "NONE", "NONE"
         return [x, y, self.__stage_dao.get_running()]
+
+    def draw_file_preview(self, check_box_click: bool, selected_file: str, canvas: Canvas, scale: int):
```
Comment (bl4szk4) [inline app/service/service.py]: I don't think that service is the best place for that method

File: eo4kry7j.dxf
Comment (bl4szk4) [inline eo4kry7j.dxf:1]: File to delete from git

File: files/ellipse_shape.dxf
Comment (bl4szk4) [inline files/ellipse_shape.dxf:1]: Same

File: square_with_circle.dxf
Comment (bl4szk4) [inline square_with_circle.dxf:1]: Same

File: zo9pge7k.dxf
Comment (bl4szk4) [inline zo9pge7k.dxf:1]: Same

File: app/files_processing/file_reading.py (hunk 2 above)
Comment (bl4szk4) [reply app/files_processing/file_reading.py]: Fixed

File: app/files_processing/file_reading.py (hunk 3 above)
Comment (bl4szk4) [reply app/files_processing/file_reading.py]: Fixed

File: app/files_processing/file_reading.py (hunk 4 above)
Comment (bl4szk4) [reply app/files_processing/file_reading.py]: Fixed

File: eo4kry7j.dxf
Comment (bl4szk4) [reply eo4kry7j.dxf:1]: Fixed

File: LASER/file_reading.py
Comment (pkarmanski) [inline LASER/file_reading.py:1]: Mateusz's file. Not needed for the application at the end.

File: LASER/lasero/file_reading.py
Comment (pkarmanski) [inline LASER/lasero/file_reading.py:1]: Mateusz's file. Not needed for the application at the end.

File: app/presentation-mateusz/presentation/components/additional_windows.py
Comment (pkarmanski) [inline app/presentation-mateusz/presentation/components/additional_windows.py:1]: Matuesz's files.

File: app/presentation/components/canvas/canvas.py [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+
+
+class Canvas(BasicCanvas):
+    def __init__(self, draw_in_canvas: callable):
+        super().__init__()
+
+        self.last_pos = None
+        self.current_pos = None
+
+        self.__scale = 1
```
Comment (pkarmanski) [inline app/presentation/components/canvas/canvas.py:16]: Check why there is a 1 assigned to `self.__scale`

File: app/service/service.py (hunk 14 above)
Comment (pkarmanski) [reply app/service/service.py]: TODO move to canvas_drawing.py

**Review by pkarmanski — APPROVED** [review]

**Pull request merged**